algorithm/area.go

```diff
@@ -0,0 +1,47 @@
+package algorithm
+
+import (
+	"math"
+
+	"github.com/spatial-go/geoos/algorithm/matrix"
+)
+
+// AreaOfMultiPolygon returns the area of a MultiPolygon geometry
+func AreaOfMultiPolygon(mp matrix.MultiPolygonMatrix) float64 {
+	area := 0.0
+	for _, polygon := range mp {
+		area += AreaOfPolygon(polygon)
+	}
+	return area
+}
+
+// AreaOfPolygon returns the area of a Polygon geometry
+func AreaOfPolygon(polygon matrix.PolygonMatrix) float64 {
+	area := 0.0
+	for i, ring := range polygon {
+		if i == 0 {
+			area += Area(ring)
+		} else {
+			area -= Area(ring)
+		}
+	}
+	return area
+}
+
+// Area returns the area of a Ring geometry.
+func Area(ring matrix.LineMatrix) float64 {
```

Comment:
Add algorithm description url

Reply:
Unified to add the algorithm description